avocado/core/future/settings.py

```diff
@@ -0,0 +1,364 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; specifically version 2 of the License.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# This code was inspired in the autotest project,
+# client/shared/settings.py
+#
+# Authors: Travis Miller <raphtee@google.com>
+#          Beraldo Leal <bleal@redhat.com>
+
+"""
+This module is a new and experimental configuration handler.
+
+This will handle both, command line args and configuration files.
+AvocadoSettings = configparser + argparser
+
+AvocadoSettings is an attempt to implement part of BP001 and concentrate all
+default values in one place. This module will read the Avocado configuration
+options from many sources, in the following order:
+
+  1. Default values (defaults.conf or defaults.py). This is a "source code"
+     file and should not be changed by the Avocado' user.
+
+  2. User/System configuration files (/etc/avocado or ~/.avocado/). This is
+     configured by the user, on a more "permanent way".
+
+  3. Command-line options parsed in runtime. This is configured by the user, on
+     a more "temporary way";
+
+ATTENTION: This is a future module, and will be moved out from this package
+soon.
+"""
+
+import configparser
+import glob
+import os
+import re
+
+from pkg_resources import resource_filename
+
+from ..settings_dispatcher import SettingsDispatcher
+from ...utils import path
+
+
+class SettingsError(Exception):
+    """
+    Base settings error.
+    """
+
+
+class ConfigFileNotFound(SettingsError):
+    """
+    Error thrown when the main settings file could not be found.
+    """
+
+    def __init__(self, path_list):
+        super(ConfigFileNotFound, self).__init__()
+        self.path_list = path_list
+
+    def __str__(self):
+        return ("Could not find the avocado config file after looking in: %s" %
+                self.path_list)
+
+
+class AvocadoSettings:
```

Comment:
One of the changes comparing this to `avocado.core.settings` is that this class got a new `Avocado` prefix.  IMO, it's unnecessary to name everything AvocadoSomething, given that it's already under an `avocado.*` namespace.  Did you think of any special reasons for this prefix?

Reply:
@clebergnu No, I don't. It was an initial design, just trying to avoid two settings. But I agree with you, I will change on v2.

Should I wait for more comments from you?

Reply:
Yes, I'm still going through the PR.

Reply:
This will be fixed on v2. Markin as resolved.